app/base/routes.py

```diff
@@ -79,12 +80,29 @@ def register():
 @blueprint.route("/account", methods=["GET", "POST"])
 @login_required
 def account():
+    from app.tasks import image_process, delete_img_obj
+
     user_id = current_user.id
     form = UpdateAccountForm()
     if form.validate_on_submit():
         if form.picture.data:
-            image = form.picture.data
-            save_profile_picture(image)
+            # delete previous profile image
+            delete_img_obj.delay(
+                s3_bucket, s3_user_img_dir, filename=current_user.photo
+            )
+
+            image = request.files["picture"]
+            folder_name = save_to_redis([image], current_user.username)
+
+            image_process.delay(folder_name, s3_user_img_dir)
+            image_names = redis_client.hgetall(folder_name)
+            image_list = [
+                f"{image_name.decode('utf-8')}.jpg" for image_name in image_names.keys()
```

Comment:
am really not comfortable why we are limiting image types to just 1 type `.jpg` here
is not that after decoding it carries its extension ?

Reply:
I refactored the code to keep the original extension of the file